core/src/test/scala/integration/kafka/api/AuthorizerIntegrationTest.scala

```diff
@@ -635,15 +746,21 @@ class AuthorizerIntegrationTest extends BaseRequestTest {
                              part: Int = part) {
     val records = new ArrayList[ConsumerRecord[Array[Byte], Array[Byte]]]()
     val maxIters = numRecords * 50
-    var iters = 0
-    while (records.size < numRecords) {
-      for (record <- consumer.poll(50).asScala) {
-        records.add(record)
+
+    val future = Future {
```

Comment:
It's a little weird here to bound this loop by the number of iterations. Could we refactor it to use `waitUntil`? For example:

```scala
    TestUtils.waitUntilTrue(() => {
        for (record <- consumer.poll(50).asScala) 
          records.add(record)
        records.size >= numRecords
    })
```

Either that or use the timeout directly in the loop instead of `maxIters`?